shuup/front/signal_handlers.py

```diff
@@ -20,25 +20,17 @@
 
 
 @receiver(context_cache_item_bumped, dispatch_uid="context-cache-item-bumped")
-def handle_context_cache_item_bumped(sender, item, shop=None, **kwargs):
-    """
-    Every time the context cache is bumped for a ShopProduct
-    we bump the context cache for template helpers for the item's shop
-    """
-    if issubclass(sender, ShopProduct):
-        if not shop and isinstance(item, int):
-            shop = ShopProduct.objects.get(id=item).shop
-        elif not shop:
-            shop = item.shop
-        context_cache.bump_cache_for_item(cache_utils.get_listed_products_cache_item(shop))
-        context_cache.bump_cache_for_item(cache_utils.get_best_selling_products_cache_item(shop))
-        context_cache.bump_cache_for_item(cache_utils.get_newest_products_cache_item(shop))
-        context_cache.bump_cache_for_item(cache_utils.get_products_for_category_cache_item(shop))
-        context_cache.bump_cache_for_item(cache_utils.get_random_products_cache_item(shop))
-        bump_product_queryset_cache()
+def handle_context_cache_item_bumped(sender, **kwargs):
+    shop_id = kwargs.get("shop_id", None)
```

Comment:
no, the old parameter was `item` which was the shop instance.
you can do like this:
```py
shop_id = kwargs.get("shop_id", None)
if not shop_id and kwargs.get("item") and isinstance(kwargs["item"], Shop):
   shop_id = kwargs["item"].pk
if not shop_id:
    shop_id = Shop.objects.first().pk
```

Reply:
well I doubt the "item" is shop id here so plan was to just delete the cache for first shop if the shop id is not passed.